### PR TITLE
Refactor, part 1

### PR DESCRIPTION
**Pull request by g7**
Let's do this incrementally so that I could also work on other things :smile: 

This pull request refactors the hwcomposer backend a bit. Most relevant changes are:

* The two hwc implementations now implement a generic interface. They don't depend on each other anymore
* The backend doesn't assume a single output anymore. Control functions (vsync, set_power_mode, etc) now work per-output
* Control functions now return a boolean with the relevant result
* Outputs can now be added and destroyed at will
* The backend now reacts on HWC2 hotplug events

More details in the various commits' descriptions.

TL;DR: If your device supports it, HDMI output/convergence works! :tada: :partying_face:

### CODE REVIEW

**Comment by g7** [comment]
Rebased on top of `feature/bullseye/startup-7c`.

**Comment by JamiKettunen** [comment]
What's the sate of this? It's been 5+ months since last push and I can't imagine it'll be fun to keep rebasing this forever on other changes done in the meanwhile

**Comment by g7** [comment]
I didn't get enough feedback on this so I decided to put that on hold for wlroots 0.12. But it will be used as a base for the upgoing migration to a more up-to-date wlroots release, which hopefully will be started in the near future.

**Comment by g7** [comment]
Closing, changes are now in latest `bookworm` branch.

**Comment by JamiKettunen** [comment]
Neat, could `bookworm` be made the default branch for this repo as well as all others applicable? That's where development happens after all ;)

**Comment by JamiKettunen** [comment]
@g7 btw should `bookworm` branch build without extra dependencies such as `android-headers` with `-Dwith-droidian-extensions=false`? that doesn't currently seem to be the case but perhaps I misunderstood the option's meaning. otherwise I'd happily supply patches :)

**Comment by g7** [comment]
> Neat, could `bookworm` be made the default branch for this repo as well as all others applicable? That's where development happens after all ;)

Indeed, done :)

> @g7 btw should `bookworm` branch build without extra dependencies such as `android-headers` with `-Dwith-droidian-extensions=false`? that doesn't currently seem to be the case but perhaps I misunderstood the option's meaning. otherwise I'd happily supply patches :)

No, `with-droidian-extensions` is currently used for (possibly Droidian-specific) quirks, at the moment only https://github.com/droidian/wlroots/commit/023a4493674af397ef91eb8dbaa569a719016359

It's possibly not needed if you don't use logind.

Personally I'd like to see the hwcomposer backend gated to a meson feature, that gets automatically enabled if `android-headers` is found. But, on the other hand, it's the only reason one is going to use this wlroots fork so I'm not sure how much can be useful...

**Comment by JamiKettunen** [comment]
Interesting, thanks for the explanation. Indeed `hwcomposer` stuff is the reason this fork exists so it's definitely not any critical issue ^^
